assets/js/live_vue/link.ts

```diff
@@ -0,0 +1,74 @@
+import { computed, defineComponent, h } from "vue"
+
+type PhxLinkAttrs =
+  | { href: string } & {
+      "data-phx-link"?: "redirect" | "patch"
+      "data-phx-link-state"?: "replace" | "push"
+    }
+
+export default defineComponent({
+  props: {
+    /**
+     * Uses traditional browser navigation to the new location.
+     * This means the whole page is reloaded on the browser.
+     */
+    href: {
+      type: String,
+      default: null,
+    },
+    /**
+     * Patches the current LiveView.
+     * The `handle_params` callback of the current LiveView will be invoked and the minimum content
+     * will be sent over the wire, as any other LiveView diff.
+     */
+    patch: {
+      type: String,
+      default: null,
+    },
+    /**
+     * Navigates to a LiveView.
+     * When redirecting across LiveViews, the browser page is kept, but a new LiveView process
+     * is mounted and its contents is loaded on the page. It is only possible to navigate
+     * between LiveViews declared under the same router
+     * [`live_session`](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.Router.html#live_session/3).
+     * When used outside of a LiveView or across live sessions, it behaves like a regular
+     * browser redirect.
+     */
+    navigate: {
+      type: String,
+      default: null,
+    },
+    /**
+     * When using `:patch` or `:navigate`,
+     * should the browser's history be replaced with `pushState`?
+     */
+    replace: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props, { attrs, slots }) {
+    const linkAttrs = computed<PhxLinkAttrs>(() => {
+      if (!props.patch && !props.navigate) {
+        return { href: props.href }
+      }
+
+      return {
+        href: props.navigate ? props.navigate : props.patch,
+        "data-phx-link": props.navigate ? "redirect" : "patch",
+        "data-phx-link-state": props.replace ? "replace" : "push",
+      }
+    })
+
+    return () => {
+      return h(
+        "a",
+        {
+          ...attrs,
+          ...linkAttrs.value,
+        },
+        slots
+      )
+    }
+  },
+})
```

Comment:
That's a very good implementation! 

One thing I'm missing are `method` related attributes. Maybe you could add them based on my implementation? It's here. Just, yours probably better because I'm not sure if shipping TS components is fine...? 

```vue
<script setup lang="ts">
import { computed } from 'vue'
interface Props {
  navigate?: string
  patch?: string
  href?: string
  replace?: boolean
  method?: string
  csrfToken?: boolean | string
}

const props = withDefaults(defineProps<Props>(), {
  replace: false,
  method: 'get',
  csrfToken: true,
})

const linkHref = computed(() => props.navigate || props.patch || props.href || '#')
</script>

<template>
  <a
    :href="linkHref"
    :data-method="method !== 'get' ? method : undefined"
    :data-csrf="method !== 'get' ? csrfToken : undefined"
    :data-to="method !== 'get' ? linkHref : undefined"
    :data-phx-link="navigate ? 'redirect' : patch ? 'patch' : undefined"
    :data-phx-link-state="(navigate || patch) && replace ? 'replace' : 'push'"
  >
    <slot></slot>
  </a>
</template>
```

Reply:
@Valian That was one of the questions that I had there, I didn't know whether to support them.

I have a question then, should we create the CSRF token ourselves just like Phoenix does? If so, we might have to read it from the live socket in `useLiveVue` if I'm not mistaken we can access it there 🤔 

WDYT?

Reply:
Another option is to accept the `csrf` token in the `getHooks` function and inject it 🤔

Reply:
About the typings, I believe there's indeed a bug introduced recently. I just handled it and released 0.5.7! 

Just, I had to make some git magic to avoid releasing recently merged test utils which needs a bit more tests :D As you said, you had a Floki error. 

About the CSRF, I'm not entirely sure. I believe it might be provided by the phonenix.js library on it's own if we pass there data-csrf = true. On the other hand, maybe let's skip data-method it for now? 🤔 It probably is not very often used ;)